### PR TITLE
Fix race condition in extension loading

### PR DESCRIPTION
**Pull request by john0312**

### CODE REVIEW

File: sites/game-client/game-client.mjs [hunk 1]
```diff
@@ -28,7 +28,11 @@ class GameClient {
     this.movementManager = movementManager;
     this.extMan = extMan;
     this.mainUI = mainUI;
+    this.avatarSelectEmitted = false;
     this.avatarSelectionPage = avatarSelectionPage;
+    this.avatarSelectionPage.setOnSelectAvatar((displayName, displayChar) => {
+      this.onSelectAvatar(displayName, displayChar);
+    });
```
Comment (ktpss95112) [inline sites/game-client/game-client.mjs:35]: `this.avatarSelectionPage.setOnSelectAvatar(this.onSelectAvatar.bind(this))`

File: sites/game-client/game-client.mjs [hunk 2]
```diff
@@ -180,6 +184,22 @@ class GameClient {
...
+
+  async onSelectAvatar(displayName, displayChar) {
+    // We need to wait for extensions to fully load before we send
+    // avatarSelect, otherwise, when gameStart event comes, some
+    // extensions might not be fully loaded, and
+    // this.extMan.startAllExtensionClient() will await at allLoadedPromise,
+    // causing race conditions, as in, some socket.io message could reach
+    // extensions before startAllExtensionClient() finishes.
+    await this.extMan.waitForAllExtLoaded();
+    if (this.avatarSelectEmitted === true) {
```
Comment (ktpss95112) [inline sites/game-client/game-client.mjs:196]: `if (this.avatarSelectEmitted) {`

File: sites/game-client/map-renderer.mjs [hunk 3]
```diff
@@ -690,6 +694,7 @@ class MapRenderer {
    * would be called to get the information for drawing.
    */
   _drawOneCharacterImage(targetCanvas, player) {
+    console.log(player);
```
Comment (ktpss95112) [inline sites/game-client/map-renderer.mjs:697]: Consider not flooding the console. Use `console.debug` or not logging this information at all.

File: sites/game-client/map-renderer.mjs [hunk 4]
```diff
@@ -317,6 +317,10 @@ class MapRenderer {
    * @param {any} renderArgs - Optional. The optional argument if renderFunction requires one.
    */
   registerCustomizedLayerToDraw(zIndex, layerName, renderFunction, renderArgs) {
+    console.assert(renderFunction !== '_drawOneCharacterImage' ||
+                     renderArgs !== null,
+                   'No render args for _drawOneCharacterImage',
+                   [zIndex, layerName, renderFunction, renderArgs]);
```
Comment (ktpss95112) [inline sites/game-client/map-renderer.mjs:323]: `renderFunction` is not necessarily `_drawOneCharacterImage`, this assert does not make sense to me.

**Comment by john0312** [comment]
Replaced by PR 367

**Pull request closed without merging**